File: Implementations_With_Java/Sorting/InsertionSort.java
public class InsertionSort {
    public static void main(String [] args)
    {
        int[] nums = {31,1,56,33,777,123,25,15,17,20,11,10,1,23,3};
        InsertionSort dz = new InsertionSort();
        dz.Sort(nums);
        dz.print(nums);
    }

    private void Sort(int[] nums){
        int temp = 0;
        int len = nums.length;
        for(int i = 1; i < len; i++)
        {
            if(nums[i] < nums[i-1])
            {
                temp = nums[i];
                nums[i] = nums[i-1];
                nums[i-1] = temp;
                i = 1;
            }
        }
    }
    private void print(int[] nums)
    {
        for(int i = 0; i < nums.length; i++)
        {
            System.out.print(nums[i] + " ");
        }
    }
}

File: Implementations_With_C/Sorting/InsertionSort.c
#include <stdio.h>
  
void swap(int *xp, int *yp){
    int temp = *xp;
    *xp = *yp;
    *yp = temp;
}
void insertionSort(int arr[] , int n){
    for (int i = 1; i < n; i++)
    {
        int toInsert = arr[i];
        int j =i-1;
        while (j>=0 && arr[j]>toInsert)
        {
            arr[j+1] = arr[j]; 
            j--;
        }
        arr[j+1] = toInsert;
    
        
    }
    
} 

void printArray(int arr[], int size)
{
    int i;
    for (i=0; i < size; i++)
        printf("%d ", arr[i]);
    printf("\n");
}
  

int main()
{
    int arr[] = {64, 25, 12, 22, 11};
    int n = sizeof(arr)/sizeof(arr[0]);
    insertionSort(arr, n);
    printf("Sorted array: \n");
    printArray(arr, n);
    return 0;
}